lib/models/amount.model.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:decimal/decimal.dart';
+
+class Amount {
+  Decimal sats = Decimal.zero;
+
+  Amount(int sats) {
+    this.sats = Decimal.fromInt(sats);
+  }
+
+  int get asSats => sats.toBigInt().toInt();
```

Comment:
❓ what does zoBigInt().toInt() do?

Reply:
It's a decimal, which is not guaranteed to fit into `int` - that's why it only offers `toBigInt`, which then offer `toInt`. I assume that the last conversion can actually fail, i.e. result in an exception. But given that we operate with `sat` amounts I don't expect them to exceed Integer's max-value; so I did not bother implementing exception handling yet 🤷‍♂️